### PR TITLE
S1 Release error handling

### PR DESCRIPTION
**Pull request by asreek**
This activity has the following changes:
1. Merged the MmeErrorCause to ERROR_CODE enum defined in the common/error_codes.h. (All the internal error codes generated by the MME across the application will have the same type.)
2. Handled abort for all states of s1 release.

### CODE REVIEW

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

**Comment by omecproject** [comment]
Can one of the admins verify this patch?

File: src/mme-app/actionHandlers/s1ReleaseActionHandlers.cpp [hunk 1]
```diff
@@ -46,40 +46,31 @@ ActStatus ActionHandlers:: send_rel_ab_req_to_sgw(SM::ControlBlock& cb)
...
+	        procCtxt->setMmeErrorCause(BEARER_CONTEXT_NOT_FOUND);return ActStatus::ABORT,
+	          "Bearer Context is NULL \n");
 
-    MmeS1RelProcedureCtxt *procCtxt = dynamic_cast<MmeS1RelProcedureCtxt*>(cb.getTempDataBlock());
-    if (procCtxt == NULL)
-    {
-        log_msg(LOG_DEBUG, "S1 Release Proc Ctxt is Null. Abort.\n");
+    	if(ue_ctxt->getS1apEnbUeId() != procCtxt->getS1apEnbUeId())
+    	{
+        	log_msg(LOG_DEBUG, "S1 Release req with wrong enb_s1ap_ue_id.\n");
```
Comment (thakurajayL) [inline src/mme-app/actionHandlers/s1ReleaseActionHandlers.cpp:71]: Will this cause rejection ? or silent drop at Mme?

**Review by thakurajayL — APPROVED** [review]

**Comment by thakurajayL** [comment]
reset this please

**Comment by thakurajayL** [comment]
@yoooou can you please see why Jenkins CI Tests are not triggered.

File: src/mme-app/actionHandlers/s1ReleaseActionHandlers.cpp (hunk 1 above)
Comment (asreek) [reply src/mme-app/actionHandlers/s1ReleaseActionHandlers.cpp:71]: This gets dropped.

**Comment by yoooou** [comment]
test this please

**Comment by yoooou** [comment]
add to whitelist

**Comment by yoooou** [comment]
> @yoooou can you please see why Jenkins CI Tests are not triggered.

@thakurajayL There was `Can one of the admins verify this patch?` posted so we'll need to manually comment something like `ok to test`

**Comment by thakurajayL** [comment]
ok to test

**Pull request merged**